### PR TITLE
Formulas

### PR DESCRIPTION
**Pull request by battermann**
- added formulas
- the UI still does not reflect the new features, yet (will be another PR)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -79,8 +79,9 @@ app.ports.startSequence.subscribe(function(seq){
   sequence = new Sequence(function(_, note){
     sampler.triggerAttackRelease(note, noteLength)
   }, seq, subdivision);
-  Transport.start()
+
   sequence.start();
+  Transport.start("+0.1")
```
Comment (jarrodu) [inline src/index.js]: What is this?

File: tests/LineTests.elm [hunk 2]
```diff
@@ -4,22 +4,74 @@ import Test exposing (..)
...
+
+
 all : Test
 all =
     describe "lines"
         [ test "C Minor pentatonic line within octave 4 should be C4, Eb4, F4, G4, Bb4" <|
             \_ ->
                 Expect.equal (fromScaleWithinRange middleOctaveRange (Scale (Note C Natural) minorPentatonic))
                     ([ Note C Natural, Note E Flat, Note F Natural, Note G Natural, Note B Flat ] |> List.map (\note -> Pitch note Octave.four))
+        , test "apply formula -2, -1, 1, -2 to C minor pentatonic from Bb3 - C6 starting with C" <|
```
Comment (jarrodu) [inline tests/LineTests.elm]: Maybe we should add some more unit testing for formulas. it seems like something that might have a few edge cases.

File: src/index.js (hunk 1 above)
Comment (battermann) [reply src/index.js]: https://github.com/Tonejs/Tone.js/wiki/Performance#scheduling-in-advance

File: tests/LineTests.elm (hunk 2 above)
Comment (battermann) [reply tests/LineTests.elm]: Oh, yes, absolutely. Maybe even some `Fuzz` tests would be nice.

**Comment by battermann** [comment]
@jarrodu Thx!
